packages/twenty-front/src/modules/views/hooks/__tests__/useViewBar.test.tsx

```diff
@@ -0,0 +1,831 @@
+import { ViewScopeInternalContext } from '@/views/scopes/scope-internal-context/ViewScopeInternalContext';
+import { currentViewFieldsScopedFamilyState } from '@/views/states/currentViewFieldsScopedFamilyState';
+import { entityCountInCurrentViewScopedState } from '@/views/states/entityCountInCurrentViewScopedState';
+import { viewEditModeScopedState } from '@/views/states/viewEditModeScopedState';
+import { viewObjectMetadataIdScopeState } from '@/views/states/viewObjectMetadataIdScopeState';
+import { viewTypeScopedState } from '@/views/states/viewTypeScopedState';
+import { ViewField } from '@/views/types/ViewField';
+import { ViewFilter } from '@/views/types/ViewFilter';
+import { ViewFilterOperand } from '@/views/types/ViewFilterOperand';
+import { ViewSort } from '@/views/types/ViewSort';
+import { ViewType } from '@/views/types/ViewType';
+
+import 'cross-fetch/polyfill';
+
+jest.mock('@/object-metadata/hooks/useMapFieldMetadataToGraphQLQuery', () => {
+  return {
+    useMapFieldMetadataToGraphQLQuery: jest.fn().mockReturnValue(() => '\n'),
+  };
+});
+
+const mockedUuid = 'mocked-uuid';
+jest.mock('uuid');
+
+(uuidv4 as jest.Mock).mockReturnValue(mockedUuid);
+
+const mocks = [
+  {
+    request: {
+      query: generateDeleteOneRecordMutation({
+        objectMetadataItem: { nameSingular: 'view' } as ObjectMetadataItem,
+      }),
+      variables: { idToDelete: mockedUuid },
+    },
+    result: jest.fn(() => ({
+      data: { deleteView: { id: '' } },
+    })),
+  },
+  {
+    request: {
+      query: gql`
+        mutation CreateOneViewField($input: ViewFieldCreateInput!) {
+          createViewField(data: $input) {
+            id
+          }
+        }
+      `,
+      variables: {
+        input: {
+          fieldMetadataId: 'id',
+          viewId: 'mocked-uuid',
+          isVisible: true,
+          size: 1,
+          position: 1,
+        },
+      },
+    },
+    result: jest.fn(() => ({
+      data: { createViewField: { id: '' } },
+    })),
+  },
+];
+const Wrapper = ({ children }: { children: React.ReactNode }) => (
+  <MemoryRouter
+    initialEntries={['/one', '/two', { pathname: '/three' }]}
+    initialIndex={1}
+  >
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <RecoilRoot>
+        <ViewScopeInternalContext.Provider
+          value={{
+            scopeId: 'viewScopeId',
+          }}
+        >
+          {children}
+        </ViewScopeInternalContext.Provider>
+      </RecoilRoot>
+    </MockedProvider>
+  </MemoryRouter>
+);
+const renderHookConfig = {
+  wrapper: Wrapper,
+};
+
+const viewBarId = 'viewBarTestId';
+
+const fieldsToPersist = [
+  {
+    fieldMetadataId: 'id',
+    label: 'label',
+    iconName: 'icon',
+    type: 'UUID',
+    isVisible: true,
+    size: 1,
+    position: 1,
+    metadata: {
+      positon: 0,
+      size: 2,
+      objectMetadataNameSingular: 'view',
+      fieldName: 'view',
+    },
+  },
+] as unknown as ViewField[];
```

Comment:
this looks suspicious, we try to avoid this in the codebase

Reply:
Fixed